docs/rollout.md

```diff
@@ -0,0 +1,175 @@
+# Rollout Strategy in Fleet
+
+Fleet uses a rollout strategy to control how apps are deployed across clusters. This feature allows you to define the order and grouping of cluster deployments, enabling controlled rollouts, and safer updates. Fleet evaluates the **Ready** status of each `BundleDeployment` to determine when to proceed to the next group. For more information, refer to [Status Fields](ref-status-fields.md).
+
+For initial deployments:
+
+* One cluster may be shown as **NotReady**.
+
+* Remaining clusters appear as **Pending**.
+
+For rollouts:
+
+* One cluster may be **NotReady**.
+
+* Remaining clusters are marked **OutOfSync** until the current one becomes ready
+
+Fleet creates up to 50 BundleDeployments at a time,  regardless of the rollout configuration. After creating a batch, Fleet evaluates the deployment status. If the current batch meets readiness criteria, Fleet proceeds to the next batch. Otherwise, it pauses the rollout.
```

Comment:
I'm torn about mentioning BundleDeployments in this documentation at all. On the one hand it, it provides some real insights, but on the other hand BundleDeployments have been considered "Fleet internal". Something that we don't expect people to know much about. Maybe we can describe that as something like "processed for the agents to be picked up" or "prepared for agents to pull and deployed" or so. I don't have a good idea right off the bat. Maybe we even want to clarify if we can speak of BundleDeployments in this context, but if we don't have any other documentation about it in our docs, perhaps we should try to phrase that differently for the time being and don't mention BundleDeployment.

@weyfonk any thoughts?